### ISSUE TITLE
"$QF: Could not create synchronized statement" when the last statement in synchronized is a throw

### ISSUE THREAD

**Issue by Kroppeb** [QuiltMC/quiltflower#163]
example source:
```java
  public void test() {
    synchronized (this) {
      throw new RuntimeException();
    }
  }
```
result:
```java
   // $QF: Could not create synchronized statement, marking monitor enters and exits
   public void test() {
      synchronized(this){} // $QF: monitorenter

      try {
         throw new RuntimeException();
      } finally {
         // $QF: monitorexit
      }
   }
```

**Comment by SuperCoder7979** [comment]
Partially fixed by 1050a6d2

**Comment by SuperCoder7979** [comment]
Fixed entirely by [3d9b2e1](https://github.com/QuiltMC/quiltflower/commit/3d9b2e157416ecbce2f7d5d58cdc116264e90c18). Thanks for the report!
